### PR TITLE
chore: Upgrade Terraform version for examples, fix links for diagrams and sources for modules

### PR DESCRIPTION
**Pull request by sebastianczech**
## Description

PR:
- upgrades Terraform version for examples
- changes links for diagrams
- fix sources for modules
- update Terraform version for modules `loadbalancer`, `virtual_machine`, `vnet` and `vnet_peering`
- update AzureRM provider for modules

## Motivation and Context

#3 

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by acelebanski** [comment]
Can you check TF and provider versions in modules as well? I see that `loadbalancer` is still 1.3 for instance.

**Comment by sebastianczech** [comment]
thanks @acelebanski - I updated min TF version for modules `loadbalancer`, `virtual_machine`, `vnet` and `vnet_peering`

**Comment by acelebanski** [comment]
> thanks @acelebanski - I updated min TF version for modules `loadbalancer`, `virtual_machine`, `vnet` and `vnet_peering`

Hi @sebastianczech, sorry I should have mentioned that earlier - I believe we shouldn't bump the TF version for `virtual_machine` module since it's for some legacy use-cases only. Migara asked not to refactor it (only thing I did in the latest PR is some descriptions changes etc. but no changes to the logic).

**Comment by sebastianczech** [comment]
@acelebanski you are right, sorry for the mistake. I restored `modules/virtual_machine` from branch `refactor-modules`.

**Comment by acelebanski** [comment]
One more thing, we also bumped AzureRM provider in some modules to `"~> 3.80"`. Most of them is still on `"~> 3.25"`. Can we bump that everywhere too?

**Comment by sebastianczech** [comment]
Done 👍

**Review by acelebanski — APPROVED** [review]
Looking good! :+1:

File: modules/natgw/.header.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ and Subnets):
 
 ```hcl
 module "natgw" {
-  source = "PaloAltoNetworks/vmseries-modules/azurerm//modules/natgw"
+  source = "PaloAltoNetworks/swfw-modules/azurerm//modules/natgw"
```
Comment (alperenkose) [inline modules/natgw/.header.md:29]: We have the following format in some of the modules, and this one as well. Shall we unify all in one format?
```
  source = "../../modules/bootstrap"
```

File: examples/standalone_vmseries/README.md
Comment (alperenkose) [inline examples/standalone_vmseries/README.md:1]: we don't have a `.header.md` file for this example?

File: modules/vmss/.header.md
Comment (alperenkose) [inline modules/vmss/.header.md:1]: do we have this file in all the modules?

File: modules/natgw/.header.md (hunk 1 above)
Comment (acelebanski) [reply modules/natgw/.header.md:29]: Hmm, I think we should use this format for any code snippets in the header.md files:
`source = "PaloAltoNetworks/swfw-modules/azurerm//modules/<module_name>"`

File: examples/standalone_vmseries/README.md
Comment (acelebanski) [reply examples/standalone_vmseries/README.md:1]: We do. What's the issue?

File: modules/vmss/.header.md
Comment (acelebanski) [reply modules/vmss/.header.md:1]: Except for `virtual_machine` module (which is out of scope), all of them have header.md files.

File: examples/standalone_vmseries/README.md
Comment (sebastianczech) [reply examples/standalone_vmseries/README.md:1]: We have but it's not changed in this PR

File: modules/natgw/.header.md (hunk 1 above)
Comment (sebastianczech) [reply modules/natgw/.header.md:29]: I updated all of `source` for modules.

**Pull request merged**